colorful/fields.py

```diff
@@ -8,21 +10,49 @@
 
 class RGBColorField(CharField):
     """Field for database models"""
-    widget = widgets.ColorFieldWidget
     default_validators = [RegexValidator(regex=forms.RGB_REGEX)]
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, colors=None, *args, **kwargs):
+        self.colors = colors
         kwargs['max_length'] = 7
         super(RGBColorField, self).__init__(*args, **kwargs)
 
     def formfield(self, **kwargs):
         kwargs.update({
             'form_class': forms.RGBColorField,
-            'widget': self.widget,
+            'widget': widgets.ColorFieldWidget(colors=self.colors),
```

Comment:
I'd keep `RGBColorField.widget` as is and simply call `self.widget(colors=self.colors)` here.